### DISCUSSION TITLE
PR created with GITHUB_TOKEN triggers on:push but does not trigger on:pull_request workflows

### DISCUSSION THREAD

**Discussion by motoshkov** [community/community#36025] [Actions]
I have a workflow which creates PR (using default GITHUB_TOKEN).
For some reason create-PR event triggers further workflow-checks having on:push but doesn't trigger workflow-checks having on:pull_request.
According to docs (https://docs.github.com/en/actions/using-workflows/triggering-a-workflow) it should not trigger any workflow.
So the question is - have status-checks workflows special handling and are they triggered by GITHUB_TOKEN created event but only on push?

**Comment by airtower-luna** [comment]
Does the workflow also create or update the branch the PR is trying to merge, or has that been pushed previously? If someone pushes a branch it's `push` event workflows will run, and if a PR is later created for that branch they will be visible as checks.
